src/plugins/intel_npu/tools/single-image-test/main.cpp

```diff
@@ -1447,9 +1632,23 @@ bool testRRMSE(const TensorMap& outputs, const TensorMap& references, size_t bat
         auto referencesIterator = references.find(tensorName);
         OPENVINO_ASSERT(referencesIterator != references.end());
 
-        std::cout << "Compare " << tensorName << " with reference" << std::endl;
-        if (!computeRRMSE(output, referencesIterator->second)) {
-            return false;
+        std::list<ov::Tensor> outputDebatchedTensors = splitBatchedTensor(output, tensorName, outputLayouts);
```

Comment:
can be move out into a templated function receiving a functor describing a particular test method, n order to get off the duplicated code. Let me check